src/ui.js

```diff
@@ -302,6 +302,52 @@ const ui = {
     //console.log("un_hookup_ui", elem.id)
   },
 
+  injectSvgNode: (node, targetNode, width, height = width) => {
+    node.setAttribute('x', 0)
+    node.setAttribute('y', 0)
+    const originalWidth = node.getAttribute('width')
+    const originalHeight = node.getAttribute('height')
+
+    targetNode.setAttribute('width', width)
+    targetNode.setAttribute('height', height)
+    targetNode.setAttribute('viewBox', `0 0 ${originalWidth} ${originalHeight}`)
+    targetNode.appendChild(node)
+  },
+
+  generateActivityItemFromTemplate: (playerAction, beforeNode, afterNode) => {
+    const templateActivityItem = document.querySelector('#template_activity_item')
+
+    templateActivityItem.content.querySelector('.activity_item_timestamp').innerText = new Date().toLocaleTimeString()
+
+    const playerNameSpan = templateActivityItem.content.querySelector('#activity_item_player_name')
+    playerNameSpan.innerText = localStorage.getItem('profile_name') || 'Unknown'
+    playerNameSpan.style.color = localStorage.getItem('profile_color')
+
+    templateActivityItem.content.querySelector('#activity_item_player_action').innerText = playerAction
+
+    const beforeNodeWrapper = templateActivityItem.content.querySelector('#activity_item_before_node')
+    const afterNodeWrapper = templateActivityItem.content.querySelector('#activity_item_after_node')
+
+    const size = 64
+    ui.injectSvgNode(beforeNode, beforeNodeWrapper, size)
+    ui.injectSvgNode(afterNode, afterNodeWrapper, size)
+
+    return document.importNode(templateActivityItem.content, true)
+  },
+
+  addActivityLogItem: (playerAction, beforeNode, afterNode) => {
```

Comment:
I split this into two functions (this and generateActivityItemFromTemplate) because I felt it read better this way. Also open to suggestions here.

Reply:
yeah i like this separation